src/app/supporttypes.jl

```diff
@@ -1,28 +1,97 @@
+struct Wildcard
+    type ::Symbol
+end
+
+JSON2.write(io::IO, wild::Wildcard; kwargs...) = Base.write(io, "[\"", wild.type, "\"]")
+
+const MATCH = Wildcard(:MATCH)
+const ALL = Wildcard(:ALL)
+const ALLSMALLER = Wildcard(:ALLSMALLER)
+
+is_wild(a::T) where {T<:Wildcard} = true
+is_wild(a) = false
+
 struct TraitInput end
 struct TraitOutput end
 struct TraitState end
 
-struct Dependency{T}
-    id ::String
+struct Dependency{Trait, IdT <: Union{String, NamedTuple}}
+    id ::IdT
     property ::String
+    Dependency{Trait}(id::T, property::String) where {Trait, T} = new{Trait, T}(id, property)
 end
 
+dep_id_string(dep::Dependency{Trait, String}) where {Trait} = dep.id
+dep_id_string(dep::Dependency{Trait, <:NamedTuple}) where {Trait} = sorted_json(dep.id)    
+
+dependency_tuple(dep::Dependency) = (id = dep_id_string(dep), property = dep.property)    
+
 const Input = Dependency{TraitInput}
 const State = Dependency{TraitState}
 const Output = Dependency{TraitOutput}
 
+"""
+    Base.==(a::Dependency, b::Dependency)
+
+We use "==" to denote two deps that refer to the same prop on
+the same component. In the case of wildcard deps, this means
+the same prop on *at least one* of the same components.
+"""
+function Base.:(==)(a::Dependency, b::Dependency) 
+    (a.property == b.property) && is_id_matches(a, b)
+end
+
+function Base.isequal(a::Dependency, b::Dependency)
+    return a == b
+end
+
+#The regular unique works using Set and will not work correctly, because in our case "equal" dependencies have different hashes
+#This implementation is not algorithmically efficient (O(n^2)), but it is quite suitable for checking the uniqueness of outputs
```

Comment:
Yep, that should be fine - this only happens when starting the server anyway, we only really care about efficiency during callback requests. Readability and simplicity are more important here.